### ISSUE TITLE
baseUrl not working properly in switchLocalePath()

### ISSUE THREAD

**Issue by hubret** [nuxt-modules/i18n#1815]
### Environment

- Operating System: `Windows_NT`
- Node Version:     `v16.16.0`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@8.11.0`
- Builder:          `vite`
- User Config:      `runtimeConfig`, `css`, `vite`, `modules`, `i18n`
- Runtime Modules:  `@nuxtjs/i18n@8.0.0-beta.8`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/github/hubret/i18nbug/tree/main/nuxt-app

https://github.com/hubret/i18nbug

### Describe the bug

the base url shows up after hitting 404 page. then you need to refresh

**Comment by ineshbose** [comment]
Handy Stackblitz link for reproduction: https://stackblitz.com/github/hubret/i18nbug/tree/main/nuxt-app

**Comment by hubret** [comment]
I added a commit that hardcodes the baseurl. first version had only set the variable in env. oops!

**Comment by kazupon** [comment]
Thank for your feedback!

Unfortunately, i18n module v8 hasn't support **nuxt baseUrl**  yet.

i18n module v8 is currently ported based on the existing v7.
Apparently nuxt baseUrl is not supported in i18n module v7.
We need to extend the existing implementation.

**Comment by hubret** [comment]
any suggestion on a workaround? some sort of string manipulation on switchLocalePath to write the baseUrl into it?

**Comment by rchl** [comment]
nuxt-i18n v7 did not include baseURL on purpose because it's not its responsibility. Vue-router does that instead. So if you are manually constructing URLs using an `a` element then you have to yourself make sure that the base URL prefix is added but if you are using `nuxt-link` then it handles base URL itself and it does not expect it to be provided explicitly.

If the module would add the base URL prefix then it wouldn't work for `nuxt-link`s. So the conclusion is: it's not a bug, use `nuxt-link` for internal links.

**Comment by kazupon** [comment]
I've checked nuxt i18n module v7.
I mentioned previous comment was wrong.
nuxt i18n module v7 and nuxt i18n module v8 support nuxt baseUrl.

As rchl mentioned, you need to use **nuxt-link** instead of a tag.
I noticed this when I read your repro code.

Thanks!

**Comment by hubret** [comment]
ah. right. initially, this wasn't working with differentDomains, but I took that out with the thought that I had isolated it to just baseUrl + switchLocalePath. And I left the `a` tags in.

~I'm going to try it with getting the current locale path and the locale domain from i18n (if its possible to access locale domain)~

~AFAICT there's no way to access locale domains directly to form a URL string for `differentDomains = true` AND `baseUrl` being set. is there any hope of including baseUrl into switchLocalePath() for the sake of differentDomains?~

Edit: I found the domains. disregard
